vignettes/join-keys.Rmd

```diff
@@ -0,0 +1,92 @@
+---
+title: "Join Keys"
+author: "NEST CoreDev"
+date: "2022-07-26"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Join Keys}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+## Overview
+`teal.data` gives `teal` applications the possibility to define individual keys per dataset and to define the relations 
+to other datasets.
+
+Each table passed into a `teal` app can have a set of keys that inform `teal` about its structure and relation
+to other tables. `teal` allows specifying:
+
+- columns consisting the primary key
+- merge keys, a concept similar to `SQL`s foreign key.
+
+Usually, an application developer needs to specify the keys manually, but in case of datasets named according
+to the `ADaM` standard, `teal` can assign the keys on its own. 
+Refer to `vignette("including-adam-data-in-teal", package = "teal")` for more information.
+
+### Primary key
+
+Using the `keys` argument to the `dataset` (or for `DDL` a `dataset_connector`) function, we can specify column(s) of the
+dataset which together uniquely identify rows in the dataset. `teal` throws an error if it finds at least two rows
+sharing all the same values of the primary key columns.
+
+```{r, eval=FALSE}
+library(teal)
```

Comment:
I think we should tweak these to not use teal and some of the text may need tweaking now this has been moved

Reply:
Updated.